### PR TITLE
For #10540: status can be a dict or a str.

### PR DESCRIPTION
**Pull request by ksallee**
Consider both cases, where status is a dict or where status is a str. Add a unittest to test both.

### CODE REVIEW

File: sg_otio/cut_diff.py [hunk 1]
```diff
@@ -457,8 +457,12 @@ def _check_and_set_changes(self):
             return
 
         # We have both a Shot and a current clip.
-        if self.sg_shot_status and self.sg_shot_status.get("code"):
-            if self.sg_shot_status["code"] in SGSettings().reinstate_shot_if_status_is:
+        if self.sg_shot_status:
+            if isinstance(self.sg_shot_status, dict) and self.sg_shot_status.get("code"):
+                sg_shot_status = self.sg_shot_status["code"]
+            elif isinstance(self.sg_shot_status, str):
+                sg_shot_status = self.sg_shot_status
```
Comment (gplsteph) [inline sg_otio/cut_diff.py:464]: else?

File: sg_otio/cut_diff.py [hunk 2]
```diff
@@ -457,8 +457,12 @@ def _check_and_set_changes(self):
             return
 
         # We have both a Shot and a current clip.
-        if self.sg_shot_status and self.sg_shot_status.get("code"):
-            if self.sg_shot_status["code"] in SGSettings().reinstate_shot_if_status_is:
+        if self.sg_shot_status:
+            if isinstance(self.sg_shot_status, dict) and self.sg_shot_status.get("code"):
+                sg_shot_status = self.sg_shot_status["code"]
+            elif isinstance(self.sg_shot_status, str):
```
Comment (gplsteph) [inline sg_otio/cut_diff.py:463]: this code is Py 3 only? Otherwise the test must be refined I think.

**Review by gplsteph — CHANGES_REQUESTED** [review]
The fix looks suspicious at first glance. In the end what is the expected value? A status or its code? It would be better to ensure that the value is consistent rather than being permissive here. This could cause problems in other parts of the code.

**Comment by ksallee** [comment]
@gplsteph as suggested I checked, and there's no reason to have this as a dict.

**Review by gplsteph — APPROVED** [review]
Thanks for checking.

**Pull request merged**